EHR_Purchasing/resources/schemas/dbscripts/postgresql/ehr_purchasing-20.000-20.001.sql

```diff
@@ -0,0 +1,176 @@
+    itemUnit            varchar(20),
+    unitDescription varchar(200),
+
+    LSID            LSIDtype,
+    container       ENTITYID NOT NULL,
+    createdBy       USERID,
+    created         timestamp,
+    modifiedBy      USERID,
+    modified        timestamp,
+
+    CONSTRAINT PK_EHR_PURCHASING_ITEM_UNITS PRIMARY KEY (rowId),
+    CONSTRAINT FK_EHR_PURCHASING_ITEM_UNITS_CONTAINER FOREIGN KEY (Container) REFERENCES core.Containers (EntityId)
+);
+
+CREATE INDEX IDX_EHR_PURCHASING_ITEM_UNITS_CONTAINER ON ehr_purchasing.itemUnits (Container);
+
+-- users associated with ehr_billing.aliases account
+CREATE TABLE ehr_purchasing.userAccountAssociations
+(
+    userId     int NOT NULL,
+    account    varchar(200) NOT NULL,
+
+    LSID       LSIDtype,
+    container  ENTITYID NOT NULL,
+    createdBy  USERID,
+    created    timestamp,
+    modifiedBy USERID,
+    modified   timestamp,
+
+    CONSTRAINT UQ_EHR_PURCHASING_USER_ACCT_ASSOCIATIONS UNIQUE (userId, account),
+    CONSTRAINT FK_EHR_PURCHASING_USER_ACCT_ASSOCIATIONS_CONTAINER FOREIGN KEY (Container) REFERENCES core.Containers (EntityId)
+);
+
+CREATE INDEX IDX_EHR_PURCHASING_USER_ACCT_ASSOCIATIONS_CONTAINER ON ehr_purchasing.userAccountAssociations (Container);
+
+-- line item status such as 'Ordered', 'Back ordered', 'Delivered' etc.
+CREATE TABLE ehr_purchasing.lineItemStatus
+(
+    rowId      serial,
+    status     varchar(100),
+
+    LSID       LSIDtype,
+    container  ENTITYID NOT NULL,
+    createdBy  USERID,
+    created    timestamp,
+    modifiedBy USERID,
+    modified   timestamp,
+
+    CONSTRAINT PK_EHR_PURCHASING_LINE_ITEM_STATUS PRIMARY KEY (rowId),
+    CONSTRAINT FK_EHR_PURCHASING_LINE_ITEM_STATUS_CONTAINER FOREIGN KEY (Container) REFERENCES core.Containers (EntityId)
+);
+
+CREATE INDEX IDX_EHR_PURCHASING_LINE_ITEM_STATUS_CONTAINER ON ehr_purchasing.lineItemStatus (Container);
+
+CREATE TABLE ehr_purchasing.purchasingRequests
+(
+    rowId            serial,
+    requestId        entityid,
+    vendorId         int,
+    account          varchar(200),
+    shippingInfoId   int,
+    justification    varchar(1000),
+    comments         varchar(1000),
+    qcStatus         int,
+
+    LSID             LSIDtype,
+    container        ENTITYID NOT NULL,
+    createdBy        USERID,
+    created          timestamp,
+    modifiedBy       USERID,
+    modified         timestamp,
+
+    CONSTRAINT PK_EHR_PURCHASING_REQUESTS PRIMARY KEY (rowId),
+    CONSTRAINT FK_EHR_PURCHASING_REQUESTS_CONTAINER FOREIGN KEY (Container) REFERENCES core.Containers (EntityId),
+    CONSTRAINT UQ_EHR_PURCHASING_REQUESTS_REQUEST_ID UNIQUE (requestId)
+);
+
+CREATE INDEX IDX_EHR_PURCHASING_REQUESTS_CONTAINER ON ehr_purchasing.purchasingRequests (Container);
+
+-- multiple line items per request. connected to purchasingRequests via requestId
+CREATE TABLE ehr_purchasing.lineItems
+(
+    rowId      serial,
+    requestId  entityid,
+    item       varchar(500),
+    itemUnitId   int,
```

Comment:
This and itemStatusId should have a FK and index, and probably be NOT NULL

Reply:
My initial thought for not having FK and index on these columns since they are "low cardinality" columns, but since we will be doing a join, having a FK (and index) makes sense - so its added now.